### PR TITLE
image-podvm: add kata components

### PR DESCRIPTION
**Pull request by katexochen**
This adds the kata components to the peerpod image and makes it usable with the upstream coco operator setup for peerpods.

To test, you can create a cluster and deploy with:

```sh
just create AKS-PEER-SNP
just get-credentials-peerpod 
just node-installer AKS-PEER-SNP
```

### CODE REVIEW

**Comment by msanft** [comment]
Note: `image_resource_group_name` is `otto-dev` for testing

**Review by Freax13 — APPROVED** [review]
I'm not qualified to review the terraform stuff, but other than that this LGTM.

File: infra/azure-peerpods/vars.tf [hunk 1]
```diff
@@ -9,3 +9,8 @@ variable "image_resource_group_name" {
 variable "subscription_id" {
   type = string
 }
+
+variable "ssh_pub_key_path" {
+  type    = string
+  default = "id_rsa.pub"
```
[the remark this answers is not in the document]
Comment (burgerdev) [reply infra/azure-peerpods/vars.tf:15]: We should consider removing support for this entirely.

[the remark this answers is not in the document]
Comment (burgerdev) [reply infra/azure-peerpods/vars.tf:15]: But the containing [`linux_profile`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#linux_profile) block is optional.

File: infra/azure-peerpods/main.tf [hunk 2]
```diff
@@ -51,23 +51,35 @@ resource "azuread_application" "app" {
...
   display_name   = local.name
-  application_id = azuread_application.app.object_id
-  audiences      = ["api://AzureADTokenExchange"]
+  application_id = azuread_application.app.id
   issuer         = azurerm_kubernetes_cluster.cluster.oidc_issuer_url
   subject        = "system:serviceaccount:confidential-containers-system:cloud-api-adaptor"
+  audiences      = ["api://AzureADTokenExchange"]
 }
 
 resource "azurerm_role_assignment" "ra_image" {
```
Comment (burgerdev) [inline infra/azure-peerpods/main.tf:85]: I believe `ra_image` is not sufficient for read permissions on the image - at least I saw corresponding failures while testing that went away after specifying the image here directly. We can sort this out later, though.

**Review by burgerdev — APPROVED** [review]

**Pull request merged**